docs/api/connectivity/bluetooth/Optimising_for_performance.md

```diff
@@ -0,0 +1,173 @@
+receive once very connection interval.
+
+The peripheral needs to receive and may transmit data if it has any. It may skip a number of these connection events
+set by `slaveLatency`. If it has no data to transmit empty packets are sent.
+
+More power is consumed if there is data to be exchanged. The exchange can continue until the next connection event would
+take place.
+
+It's worth considering if keeping the connection active is worth it. Connection in BLE is extremely fast and if you plan
+to only send a quick burst of data every minute it is better to connect and disconnect each time.
+
+The cost of the connection is proportionate to the negotiable connection interval. This can be set during `connect` or
+later through `updateConnectionParameters`. The lower the interval the more often radio is active. This is especially
+important for the peripheral which needs to enable the radio to receive packets.
+
+This can be further helped through setting a high `slaveLatency` parameter. This allows the peripheral to skip
+connection events and save power not just by not sending any packets but by not even listening. This is not free for
+central as it increases latency of data transmission from central to peripheral. Central may have to attempt sending
+data multiple times before the peripheral accepts the transmission. The peripheral may send data at any connection event
+as the central must listen after every transmission.
+
+### Advertising and scanning
+
+Power draw during advertising affected by:
+- the advertising interval - lower interval uses more power,
+- amount of data sent,
+- number of channels used - each advertising event is sent by default to three channels which you can limit to 2 or 1,
+- whether extended advertising is used - this will send additional packets on regular channels,
+- whether the type is connectable or scannable - it means the advertiser needs to listen on the radio after each
+  advertisement for potential connection of scan requests.
+
+
+               PERIPHERAL
+                ┌────┐           advertising interval              ┌────┐
+     channel 37 │adv │◄───────────────────────────────────────────►│adv │
+                └────┘                                             └────┘
+                      ┌────┐                                             ┌────┐
+     channel 38       │adv │                                             │adv │
+                      └────┘                                             └────┘
+                            ┌────┐                                             ┌────┐
+     channel 39             │adv │                                             │adv │
+                            └────┘                                             └────┘
+
+     non-advertising              ┌────────────────────┐
+     channel                      │extended advertising│
+     (indicated in regular        └────────────────────┘
+      advertising payload)
+
```

Comment:
Render of the diagram in markdown is unreadable for me :(.

Reply:
Here is what I see:

<img width="1054" alt="Screenshot 2021-07-12 at 13 55 02" src="https://user-images.githubusercontent.com/409941/125290973-cd6b1780-e318-11eb-9e59-14009da6e4e7.png">

Reply:
don't know why github doesn't follow the 4 space formatting, added ```